cql3/selection/selectable.cc

```diff
@@ -76,27 +76,6 @@ selectable::writetime_or_ttl::to_string() const {
     return format("{}({})", _is_writetime ? "writetime" : "ttl", _id->to_string());
 }
 
-shared_ptr<selector::factory>
```

Comment:
I don't know where to ask this so I'll ask here:
If we prepare a statement which mentions a UDF name, and later we drop this UDF and then try to execute the prepared statement - what will happen? Do we have a test for this case?

Reply:
We should drop the function from the cache. We probably don't, the cached statement will hold on to the shared pointer and keep it alive until it's evicted.

Reply:
```c++
void query_processor::migration_subscriber::on_drop_keyspace(const sstring& ks_name) {
    remove_invalid_prepared_statements(ks_name, std::nullopt);
}

void query_processor::migration_subscriber::on_drop_column_family(const sstring& ks_name, const sstring& cf_name) {
    remove_invalid_prepared_statements(ks_name, cf_name);
}

void query_processor::migration_subscriber::on_drop_user_type(const sstring& ks_name, const sstring& type_name) {
}

void query_processor::migration_subscriber::on_drop_function(const sstring& ks_name, const sstring& function_name) {
    log.warn("{} event ignored", __func__);
}

void query_processor::migration_subscriber::on_drop_aggregate(const sstring& ks_name, const sstring& aggregate_name) {
    log.warn("{} event ignored", __func__);
}

void query_processor::migration_subscriber::on_drop_view(const sstring& ks_name, const sstring& view_name) {
    remove_invalid_prepared_statements(ks_name, view_name);
}
```